framework/i18n/Formatter.php

```diff
@@ -168,6 +169,21 @@ class Formatter extends Component
      * @see http://php.net/manual/en/class.intlcalendar.php
      * @since 2.0.7
      */
+
+    public $dateRangeFormat = [
+                'showTime' => false,
+                'year' => 'Y',
+                'month' => 'M',
+                'day' => 'j',
+                'time' => 'H:i',
+                'templates' => [
+                    'notDiff' => '{from_time} {from_day} {from_month} {from_year}',
+                    'diffYear' => '{from_time} {from_day} {from_month} {from_year} - {to_time} {to_day} {to_month} {to_year}',
+                    'diffMonth' => '{from_time} {from_day} {from_month} - {to_time} {to_day} {to_month} {to_year}',
+                    'diffDay' => '{from_time} {from_day} - {to_time} {to_day} {to_month} {to_year}',
+                    'diffTime' => '{from_time} - {to_time} {to_day} {to_month} {to_year}'
+                ]
+            ];
```

Comment:
There is formatting information about this in intl ICU data. I think we should take advantage of that:

e.g. http://intl.rmcreative.ru/tables?locale=de search for `intervalFormats` on that page.